### PR TITLE
Modified the existing Effect directive in sample code of docs

### PR DESCRIPTION
**Pull request by jaferkhan**
# Pull Request Checklist
* [x] Have you read through the [contributor guidelines](./CONTRIBUTING.md)?
* [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [x] Have you updated the documentation?
* [ ] Have you added tests for any changed functionality?

## Purpose
Replaces the existing Effect directive in sample code, which wrongfully persisted the command inside an instance of `CommandHandler`

## References
References #28115

### CODE REVIEW

**Comment by johanandren** [comment]
OK TO TEST

File: akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala [hunk 1]
```diff
@@ -200,9 +200,7 @@ object BasicPersistentBehaviorCompileOnly {
           emptyState = State(),
           commandHandler = CommandHandler.command { cmd =>
             context.log.info("Got command {}", cmd)
-            Effect.persist(cmd).thenRun { state =>
-              context.log.info("event persisted, new state {}", state)
-            }
+            Effect.none
```
Comment (johanandren) [inline akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala]: Probably has the same in the corresponding Java sample that needs an update as well?

Comment (jaferkhan) [reply akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala]: I am glad that you pointed this out but interestingly no commandHandler() method exists on the Java side. 

You may have a look at it here: https://doc.akka.io/docs/akka/2.6/typed/persistence.html?#accessing-the-actorcontext

Comment (johanandren) [reply akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala]: Ah, thanks for checking!

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala (hunk 1 above)
Comment (jaferkhan) [reply akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala]: You're most welcome :)

**Review by helena — APPROVED** [review]
Thanks for the change, LGTM

**Comment by superleo-cn** [comment]
Verified.

**Review by superleo-cn — COMMENTED** [review]
Verified.

**Pull request merged**